apps/meteor/app/lib/server/lib/notifyListener.ts

```diff
@@ -501,3 +503,263 @@ export async function notifyOnUserChangeById({ clientAction, id }: { id: IUser['
+	clientAction: ClientAction = 'updated',
+): Promise<void> {
+	if (!dbWatchersDisabled) {
+		return;
+	}
+
+	const subscriptions =
+		clientAction === 'removed'
+			? Subscriptions.trashFind({ rid, 'u._id': uid }, { projection: subscriptionFields })
+			: Subscriptions.findByUserIdAndRoomIds(uid, [rid], { projection: subscriptionFields });
+
+	if (!subscriptions) {
+		return;
+	}
+
+	for await (const subscription of subscriptions) {
+		void api.broadcast('watch.subscriptions', { clientAction, subscription });
+	}
+}
+
+export async function notifyOnSubscriptionChangedById(id: ISubscription['_id'], clientAction: ClientAction = 'updated'): Promise<void> {
+	if (!dbWatchersDisabled) {
+		return;
+	}
+
+	const subscription = clientAction === 'removed' ? await Subscriptions.trashFindOneById(id) : await Subscriptions.findOneById(id);
+
+	if (!subscription) {
+		return;
+	}
+
+	void api.broadcast('watch.subscriptions', { clientAction, subscription });
+}
+
+export async function notifyOnSubscriptionChangedByRoomIdExcludingUserIds(
+	rid: ISubscription['rid'],
+	uids: ISubscription['u']['_id'][],
+	clientAction: ClientAction = 'updated',
+): Promise<void> {
+	if (!dbWatchersDisabled) {
+		return;
+	}
+
+	const subscriptions = Subscriptions.findByRoomIdExcludingUserIds(rid, uids, { projection: subscriptionFields });
+
+	for await (const subscription of subscriptions) {
+		void api.broadcast('watch.subscriptions', { clientAction, subscription });
+	}
+}
+
+export async function notifyOnSubscriptionChangedByUserPreferences(
+	uid: ISubscription['u']['_id'],
+	notificationOriginField: keyof ISubscription,
+	originFieldNotEqualValue: 'user' | 'subscription',
+	clientAction: ClientAction = 'updated',
+): Promise<void> {
+	if (!dbWatchersDisabled) {
+		return;
+	}
+
+	const subscriptions = Subscriptions.findByUserPreferences(uid, notificationOriginField, originFieldNotEqualValue, {
+		projection: subscriptionFields,
+	});
+
+	for await (const subscription of subscriptions) {
+		void api.broadcast('watch.subscriptions', { clientAction, subscription });
+	}
+}
+
+export async function notifyOnSubscriptionChangedByRoomId(
+	rid: ISubscription['rid'],
+	clientAction: ClientAction = 'updated',
+): Promise<void> {
+	if (!dbWatchersDisabled) {
+		return;
+	}
+
+	const subscriptions =
+		clientAction === 'removed'
+			? Subscriptions.trashFind({ rid }, { projection: subscriptionFields })
+			: Subscriptions.findByRoomId(rid, { projection: subscriptionFields });
+
+	if (!subscriptions) {
+		return;
+	}
+
+	for await (const subscription of subscriptions) {
+		void api.broadcast('watch.subscriptions', { clientAction, subscription });
+	}
+}
+
+export async function notifyOnSubscriptionChangedByToken(token: string, clientAction: ClientAction = 'updated'): Promise<void> {
+	if (!dbWatchersDisabled) {
+		return;
+	}
+
+	const subscriptions =
+		clientAction === 'removed'
+			? Subscriptions.trashFind({ 'v.token': token }, { projection: subscriptionFields })
```

Comment:
I don't know how often we'd notify removed records, maybe in this particular case we'd call more, since it can be more common to omnichannel use cases, but this comment may be relevant to every other notify of removed records: the "trash collection" doesn't have the same indexes as the "original" collections, and that's because there is only a single trash collection to store records from all other collections... so if the collection is somewhat full, having to query by different fields other than `_id` can be concerning.. 

the alternative would be to change wherever we call "notify removed" to first fetch all records that will be removed, remove them, and then notify, without having to perform a find in the trash collection.

not sure though, I'm open to discussion about this.